Sources/App/Setup/repositories.swift

```diff
@@ -0,0 +1,7 @@
+import Vapor
+
+public func setupRepositories(_ config: inout Config, _ env: inout Environment, _ services: inout Services) {
+    services.register(TodoRepository.self) { _ -> TodoRepository in
```

Comment:
I think this should be returning a `SQLiteTodoRepository`, the idea being that in your tests you can create a `MemoryTodoRepository` and register that one for that environment. Then the overarching type can still be a common protocol, `TodoRepository`. This also makes it possible to only ask for a `TodoRepository` in the routes and get back whichever was registered - memory or sqlite depending on the state that the project is being run in.